packages/format-library/src/text-color/inline.js

```diff
@@ -44,9 +44,10 @@ function parseCSS( css = '' ) {
 
 function parseClassName( className = '', colorSettings ) {
 	return className.split( ' ' ).reduce( ( accumulator, name ) => {
-		const match = name.match( /^has-([^-]+)-color$/ );
-		if ( match ) {
-			const [ , colorSlug ] = name.match( /^has-([^-]+)-color$/ );
+		if ( name.startsWith( 'has-' ) && name.endsWith( '-color' ) ) {
```

Comment:
There should be a comment here explaining that `colorSlug` may be more than one word long, lest someone reintroduce a naive regular expression.